foca/foca.py

```diff
@@ -0,0 +1,97 @@
+""" Definition of an entry-point function for setting up and initializing a
+FOCA based microservice."""
+
+from foca.api.register_openapi import register_openapi
+from foca.config.log_config import configure_logging
+from foca.database.register_mongodb import register_mongodb
+from foca.errors.errors import register_error_handlers
+from foca.factories.connexion_app import create_connexion_app
+from foca.factories.celery_app import create_celery_app
+from foca.security.cors import enable_cors
+from foca.config.config_handler import Config
+from connexion import App
+from typing import Mapping, Any
+
+
+def foca(config: Mapping[str, Any]

```

Comment:
Sorry, guess I should have clearer here: the arg is going to be one or more _file paths_ to YAML config files (or possibly one or more environment variables, but for now, let's keep it simple, so just supporting file paths is fine). You will then call the config parser which will process and return the app with the config attached as an instance of the config model class that @kushagra189 is writing. This flow makes it even easier for the client. (Although I have to say that I'm not 100% convinced with that, so don't flame me if that turns out to be a bad call later on. Pretty sure it's gonna be fine though).

Reply:
So, eventually the input will be of type `Iterable[str]`, right?
For creating the `Config` model class instance for each configuration file, I only need to invoke `YAMLConfigParser`, but isn't that the only thing that's going to get returned? (and not the whole `app` instance with its configuration attached). Shouldn't I *then* call `foca.factories.connexion_app.create_connexion_app()` with the aforementioned `Config` instance as input?

Reply:
Sorry for the confusion, it's because I was just projecting more or less how the config would/should behave at a time when it wasn't yet implemented. Also, I was definitely confused about the point when the app is created. Anyway, after roughly implementing the config parsing last night, I can confirm:

You only pass a _single_ config file path, so expected input is `str`. We could implement multiple paths to allow users to pass multiple partial files to manage their config, but last night I thought that's feature creep, and even though I had made use of that multi config file functionality in cwl-WES in the past (for different deployment scenarios), it turned out to be confusing and harder to maintain for little gain (as k8s deployment has better ways of dealing with that).

And yes, you receive an instance of object in return that you then pass when you create the app.

Reply:
Perfect.